config.yml

```diff
@@ -1766,6 +1825,7 @@ nodes:
         type: location?
       - name: value
         type: node
+        kind: non-void expression # TODO: not sure if this is correct, e.g. defined?(break/return/redo/next/yield) is allowed, but are those "void expressions"?
```

Comment:
Are `break/return/redo/next/yield` void expressions?
If yes, we should type this as `Node` to be exact (although the difference is only for documentation purpose in this file).

As a note, `defined?(BEGIN{})` is an error, but the PreExecutionNode ends up in the AST. So technically `on error: Node` but that seems needlessly complex.

Reply:
According to pm_check_value_expression() they seem to be, except `yield`, which makes sense `yield` doesn't unwind and it returns a value (my mistake in the comment).

Reply:
I changed this to `kind: Node # More than non-void expression as defined?(return) is allowed, yet defined?(BEGIN{}) is SyntaxError`